tutorials/dapt-curation/code/docbuilder.py

```diff
@@ -0,0 +1,486 @@
+                    if example:
+                        yield split_meta(example)
+                        example = []
+                else:
+                    example.append(line.strip())
+
+            if example:
+                yield split_meta(example)
+
+
+class WikitxtExtractor(DocumentExtractor):
+    def extract(self, content: str) -> Tuple[Set, str]:
+        # No metadata for the text, just the content.
+        return {}, content
+
+
+class GitHubDownloader(DocumentDownloader):
+    """
+    A class for downloading repositories from GitHub.
+    """
+
+    def __init__(self, github_root_dir: str):
+        """
+        Initializes the DocBuilder object.
+
+        Args:
+            github_root_dir: The root directory for GitHub repositories.
+        """
+        super().__init__()
+        # The path under which the repositories will be cloned.
+        self.clone_root_dir = os.path.join(github_root_dir, "repos")
+        os.makedirs(github_root_dir, exist_ok=True)
+        os.makedirs(self.clone_root_dir, exist_ok=True)
+
+    def download(self, url: str) -> str:
+        """
+        Download a repository as a zip file.
+
+        Args:
+            url (str): The URL of the repository.
+
+        Returns:
+            str: The path to the downloaded zip file, or None if the download failed.
+        """
+        repo_name = os.path.basename(url)
+        zip_file = os.path.join(self.clone_root_dir, repo_name + ".zip")
+
+        if os.path.exists(zip_file):
+            print(f"Repository '{repo_name}' already exists, skipping download.")
+            return zip_file
+
+        # Try the common branch names first. A better way to do this would be to
+        # query the GitHub API to get the default branch, but that is subject to rate limits.
+        success = False
+
+        for branch in ["master", "main"]:
+            zip_url = f"https://github.com/{url}/archive/refs/heads/{branch}.zip"
+
+            # Send a GET request to the URL
+            response = requests.get(zip_url)
+
+            # Check if the request was successful
+            if response.status_code == 200:
+                # Write the content of the response to a file
+                with open(zip_file, "wb") as file:
+                    file.write(response.content)
+
+                # No need to try other branches
+                success = True
+                break
+
+        if not success:
+            print(
+                f"Failed to clone repository '{repo_name}' from '{url}' (error code {response.status_code})."
+            )
+            return None
+
+        return zip_file
+
+
+class GitHubIterator(DocumentIterator):
+    """
+    GitHub document iterator. Will go through the files and parse the supported ones.
+    """
+
+    # Mapping from file extensions to categories.
+    # Will also be used to to ignore irrelevant files.
+    SUPPORTED_EXTENSIONS_TO_CATEGORY = {
+        ".vx": "Viva",
```

Comment:
Let's remove `.vx` and `.vxh`  as those are proprietary extensions that won't exist in the GitHub repos.

Reply:
done